### PR TITLE
XML Documentation Fixes - Batch 4

### PR DESCRIPTION
**Pull request by benrr101**
**Description**: This is batch 4 of the documentation changes to chip away at #2013 

**Testing**: I went through each of these to spot check for formatting issues. It might not be perfect but I'm pretty happy with it.

### CODE REVIEW

File: doc/snippets/Microsoft.Data.SqlClient/SqlCommandBuilder.xml [hunk 1]
```diff
@@ -1,439 +1,530 @@
...
+    </DeriveParameters>
+    <Dispose>
+      <param name="disposing">
+        To be added.
+      </param>
+      <summary>
+        To be added.
+      </summary>
+      <remarks>
+        To be added.
```
Comment (cheenamalhotra) [inline doc/snippets/Microsoft.Data.SqlClient/SqlCommandBuilder.xml:183]: Were you going to rewrite these remarks based on available formatted CDATA?

Comment (benrr101) [reply doc/snippets/Microsoft.Data.SqlClient/SqlCommandBuilder.xml:183]: I would, but there isn't any in main...
image

I've stripped out <remarks>To be added</remarks> for ones that have other comments (basically under the validation that if we haven't needed remarks up to now, we probably won't ever need them). But if there aren't any comments other than to be added, then I opted to keep it 1) so we have some comments and 2) so we can maybe one day go through and add some. But, writing them now is outside the scope of these PRs.

File: doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml [hunk 2]
```diff
@@ -1,1340 +1,3326 @@
...
+        Gets or sets the access token callback for the connection.
+      </summary>
+      <value>
+        The Func that takes a <see cref="T:SqlAuthenticationParameters" /> and <see cref="T:System.Threading.CancellationToken" /> and returns a <see cref="T:Microsoft.Data.SqlClient.SqlAuthenticationToken" />.
+      </value>
+      <example>
+        <para>
+          The following example demonstrates how to define and set an <see cref="P:Microsoft.Data.SqlClient.SqlConnection.AccessTokenCallback" />.
+        </para>
+        <!-- SqlConnection_AccessTokenCallback -->
```
[the remark this answers is not in the document]
Comment (benrr101) [reply doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml:280]: Gotcha. I updated that specific one - but are there other ones that need to be updated?

File: doc/snippets/Microsoft.Data.SqlClient/SqlCommandBuilder.xml (hunk 1 above)
Comment (cheenamalhotra) [reply doc/snippets/Microsoft.Data.SqlClient/SqlCommandBuilder.xml:183]: Sure, thanks!

File: src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs [hunk 3]
```diff
@@ -865,6 +865,7 @@ public void RegisterColumnEncryptionKeyStoreProvidersOnConnection(System.Collect
         [System.ComponentModel.DesignerSerializationVisibilityAttribute(0)]
         internal string SQLDNSCachingSupportedStateBeforeRedirect { get { throw null; } }
 
+        /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml' path='docs/members[@name="SqlConnection"]/System.ICloneable.Clone/*' />
         object System.ICloneable.Clone() { throw null; }
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs:869]: Surprised it didn't error out for missing doc comment yet 🤔

**Review by cheenamalhotra — APPROVED** [review]

**Review by mdaigle — APPROVED** [review]
LGTM

File: doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (David-Engel) [reply doc/snippets/Microsoft.Data.SqlClient/SqlConnection.xml:280]: That's the only one I see.

File: src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs (hunk 3 above)
Comment (benrr101) [reply src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs:869]: I think it only errors out if there was a change to the code? There's a handful I've noticed along the way that don't have documentation but still build.

**Pull request merged**